icechunk-python/src/repository.rs

```diff
@@ -578,12 +594,20 @@ impl PyRepository {
         tag: Option<String>,
         snapshot_id: Option<String>,
     ) -> PyResult<PyAsyncGenerator> {
-        let repo = Arc::clone(&self.0);
         // This function calls block_on, so we need to allow other thread python to make progress
         py.allow_threads(move || {
             let version = args_to_version_info(branch, tag, snapshot_id, None)?;
             let ancestry = pyo3_async_runtimes::tokio::get_runtime()
-                .block_on(async move { repo.ancestry_arc(&version).await })
+                .block_on(async move {
+                    let (snapshot_id, asset_manager) = {
+                        let lock = self.0.read().await;
+                        (
+                            lock.resolve_version(&version).await?,
+                            Arc::clone(lock.asset_manager()),
```

Comment:
nit: you could do the `Arc::clone` once the lock is released.

Reply:
How? We need the lock to get the reference to the asset manager right?

Reply:
lock -> get the `&Arc<AssetManager>` -> unlock -> Clone

Reply:
Aha, I didn't know you could keep a reference after locking

Reply:
Ah, I didn't really think about that... I'm not sure now

Reply:
Yeah i have to clone from a scope where the lock is held. Which i handled by making an inner scope so it is released immediately after cloning the asset manager